ribbon/contract.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#----------------------------------------------------------------------------
+# Created By: Steven (steven@ribbon.finance)
+# Created Date: 04/04/2022
+# version ='0.01'
+# ---------------------------------------------------------------------------
+''' Abstract class for contract factory '''
+# ---------------------------------------------------------------------------
+
+# ---------------------------------------------------------------------------
+# Imports
+# ---------------------------------------------------------------------------
+from web3 import Web3
+import json
+
+# ---------------------------------------------------------------------------
+# Contract Factory
+# ---------------------------------------------------------------------------
+class ContractFactory:
+  '''
+  Object to create connection to a contract
+
+  Args:
+      rpc (str): Json RPC address to connect
+      address (str): Contract address
+      abi (dict): Contract ABI location
+
+  Attributes:
+      address (str): Contract address
+      abi (dict): Contract ABI
+      w3 (object): RPC connection instance
+      contract (object): Contract instance
+  '''
+  def __init__(self, rpc: str, address: str, abi: dict) -> None:
```

Comment:
It seems `rpc` would be like `"https://kovan.infura.io/v3/" + TOKEN`.

Should we have that become 
```python
INFURA_URL = "https://kovan.infura.io/v3/"

...

def __init__(self, rpc_token: str, address: str, abi: dict, base_rpc_url: str = INFURA_URL) -> None:
     rpc = base_rpc_url + rpc_token
```
and have the ability to change it, but have a comfortable default.

Reply:
The URL depends on the chain. For example, mainnet uses `"https://mainnet.infura.io/v3/"`. I much prefer not to give any default so user is aware which chain they are interacting with.

Reply:
Should we still split into `rpc_chain_url` and `rpc_token`?

Reply:
Typically we don't really use multiple tokens for each URL. Any particular use case you're thinking of?

Reply:
Alright, changed